### ISSUE TITLE
Require Moengage Module integration for Push Notifications, In App Notifications and events

### ISSUE THREAD

**Issue by mehsaandev** [EnsembleUI/ensemble#1773]
Require the following Modules from Moengage:
- In-App Notification Banners
- Push Notifications
- Deep Linking to navigate on a Specific Screen when taping Notification.
- Handle Moengage Events

**Comment by kmahmood74** [comment]
@mehsaandev what does this mean - `Deep Linking to navigate on a Specific Screen when taping Notification.` what will the deep link look like and which screen should it go with what payload? Please be specific with examples instead of a super generic statement

**Comment by mehsaandev** [comment]
- Deep Linking for Ensemble:
Following is the dashboard section in Moengage for deep linking. 
I can add key-value pairs for deep linking in the notifications.
For navigation Please consider `screen` key to get the target screen to navigate on.

![image](https://github.com/user-attachments/assets/3c28bae6-e1f9-4da5-879c-827f0681bb1e)

CC: @TheNoumanDev

**Comment by kmahmood74** [comment]
@mehsaandev have you confirmed with the customer that they will be creating special notifications with the screen names which means that their existing notification payloads will not work anymore? Are we sure we want to impose this?

**Comment by mehsaandev** [comment]
> @mehsaandev have you confirmed with the customer that they will be creating special notifications with the screen names which means that their existing notification payloads will not work anymore? Are we sure we want to impose this?

@kmahmood74  Yes I've asked the customer and they are totally fine with this solution.
